src/main/webapp/app/modules/canvas/visualization/canvas.tsx

```diff
@@ -143,10 +97,49 @@ const Canvas = (props: VisualizationProp) => {
     }
   }, []);
 
+  useEffect(() => {
+    if (props.isSocketConnected) {
+      VisualMetadataContainerAdd(props.visualmetadata?.visualMetadataSet);
+      loadVisualization();
+    }
+  }, [props.isSocketConnected]);
+
+  useEffect(() => {
+    if (props.visualData) {
+      if (props.visualData?.headers?.request === 'filters') {
```

Comment:
can you put this condition inside receiveSocketResponse. this should be for all visuaslization filters.let me know if you need a call

Reply:
done